File: package/main/src/Date/subDuration.ts
import { addDuration } from "./addDuration";
import type { DurationUnit } from "./durationUnit";

/**
 * Subtracts a duration from a date and returns a new Date.
 * Equivalent to addDuration(date, -amount, unit).
 *
 * @param {Date} date - Base date
 * @param {number} amount - Amount to subtract
 * @param {DurationUnit} unit - Unit of the amount
 * @returns {Date} A new Date with the duration subtracted
 * @example
 * subDuration(new Date("2025-03-31"), 1, "M"); // 2025-02-28
 */
export const subDuration = (
  date: Date,
  amount: number,
  unit: DurationUnit,
): Date => addDuration(date, -amount, unit);

File: package/umt_python/src/date/sub_duration.py
from datetime import datetime

from .add_duration import add_duration


def sub_duration(date: datetime, amount: int | float, unit: str) -> datetime:
    """
    Subtract a duration from a date and return a new datetime.

    Equivalent to ``add_duration(date, -amount, unit)``.

    Args:
        date: Base date
        amount: Amount to subtract
        unit: Unit of the amount
            ("ms"|"s"|"m"|"h"|"d"|"w"|"M"|"y")

    Returns:
        A new datetime with the duration subtracted

    Example:
        >>> from datetime import datetime
        >>> sub_duration(datetime(2025, 3, 31), 1, "M")
        datetime.datetime(2025, 2, 28, 0, 0)
    """
    return add_duration(date, -amount, unit)
